matter/wifi/wf200/wf200_init.c

```diff
@@ -93,26 +101,38 @@ sl_status_t sl_wfx_host_disable_spi(void);
 #endif
 
 /****************************************************************************
+ * @fn  sl_status_t wfx_soft_init(void)
+ * @brief
  * WFX FMAC driver host interface initialization
+ * @param[in] None
+ * @returns Returns SL_STATUS_OK if successful,
+ *          SL_STATUS_FAIL otherwise
  *****************************************************************************/
 sl_status_t wfx_soft_init(void)
 {
   EFR32_LOG("WF200:Soft Init");
-  if ((wfx_event_Q == NULL) && ((wfx_event_Q = xQueueCreate(SL_WFX_EVENT_LIST_SIZE, sizeof(uint8_t))) == NULL)) {
+  if ((wfx_event_Q == NULL) && ((wfx_event_Q = xQueueCreateStatic(SL_WFX_EVENT_LIST_SIZE, sizeof(uint8_t), sWfxEventQueueBuffer, &sWfxEventQueueStruct)) == NULL)) {
     return SL_STATUS_FAIL;
   }
 
-  if ((wfx_wakeup_sem == NULL) && ((wfx_wakeup_sem = xSemaphoreCreateBinary()) == NULL)) {
+  if ((wfx_wakeup_sem == NULL) && ((wfx_wakeup_sem = xSemaphoreCreateBinaryStatic(&xWfxWakeupSemaBuffer)) == NULL)) {
```

Comment:
In xSemaphoreCreateBinaryStatic  If the semaphore is created successfully then a the returned value is a handle by which the semaphore can be referenced and a NULL otherwise.

this is error check condition if semaphore is not created or not, if not created then stop right there.

Reply:
I removed the first check condition of checking with NULL.

And initialized the variable with NULL in starting.